### PR TITLE
Move Users, Roles & Required Signoffs into a dedicated section

### PR DESCRIPTION
**Pull request by mozbhearsum**
This is mainly a layout/nav change to put User, Role, and Required Signoff management into a dedicated "Settings" section, since they are not part of day-to-day operations. I didn't move the existing `Signoffs` section in, because I wasn't sure if you had anything in progress on it, and didn't want to cause merge conflicts. I'm happy to move it in if that's helpful, though.

If there's a better or simpler way to do this than `react-sidenav` I'm happy to switch it up - I just chose it because it seemed to be reasonably well mantained, and easy enough to work.

This could probably do with some styling improvements later, but it's functional for now.

### CODE REVIEW

**Comment by helfi92** [comment]
@mozbhearsum What do you think about using a menu instead of adding a side nav? It would be a menu similar to the sign-out button when a user is logged in:

Screen Shot 2019-05-15 at 8 47 24 AM

This will make sure all pages are visible to the user from the main navigation menu at the top. A new user trying to navigate to the "Required Signoffs" view for example will find the view quickly without having to guess where it is.

**Comment by mozbhearsum** [comment]
That seems like a pretty good idea, I'll update for that.

**Comment by mozbhearsum** [comment]
I think I've got this working. I ended up moving the existing `Signoffs` views into `RequiredSignoffs` as well (it looked weird to have both at the top level).

**Comment by mozbhearsum** [comment]
Hmm, it looks like using a `Link` as the `MenuItem` component is causing an issue:
> Warning: Failed prop type: Invalid prop `component` supplied to `ButtonBase`. Expected an element type that can hold a ref. Did you accidentally provide a plain function component instead? For more information see https://next.material-ui.com/guides/composition/#caveat-with-refs

Going to try to get that fixed up.

**Comment by helfi92** [comment]
> Hmm, it looks like using a Link as the MenuItem component is causing an issue:

I suspect this is an issue with the alpha version of `material-ui`. I think it's safe to ignore for now until we upgrade to the latest version `4.0.0-beta.2`.

File: src/components/Dashboard/SettingsMenu.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import React, { useState, Fragment } from 'react';
+import { makeStyles } from '@material-ui/styles';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import IconButton from '@material-ui/core/IconButton';
+import Settings from '@material-ui/icons/Settings';
```
Comment (helfi92) [inline src/components/Dashboard/SettingsMenu.jsx]: For icons, let's use `mdi-react` which has all material icons + community icons.

```jsx
import SettingsOutlineIcon from 'mdi-react/SettingsOutlineIcon';
```

File: src/components/Dashboard/SettingsMenu.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  return (
+    <Fragment>
+      <IconButton
+        className={classes.avatar}
+        aria-haspopup="true"
+        aria-controls="user-menu"
+        aria-label="user menu"
+        onClick={handleMenuOpen}>
+        <Settings />
```
Comment (helfi92) [inline src/components/Dashboard/SettingsMenu.jsx]:
```diff
diff --git a/src/components/Dashboard/SettingsMenu.jsx b/src/components/Dashboard/SettingsMenu.jsx
index aa5580b..39e459d 100644
--- a/src/components/Dashboard/SettingsMenu.jsx
+++ b/src/components/Dashboard/SettingsMenu.jsx
@@ -3,7 +3,7 @@ import { makeStyles } from '@material-ui/styles';
 import Menu from '@material-ui/core/Menu';
 import MenuItem from '@material-ui/core/MenuItem';
 import IconButton from '@material-ui/core/IconButton';
-import Settings from '@material-ui/icons/Settings';
+import SettingsOutlineIcon from 'mdi-react/SettingsOutlineIcon';
 import Link from '../../utils/Link';
 
 const useStyles = makeStyles(theme => ({
@@ -11,6 +11,10 @@ const useStyles = makeStyles(theme => ({
     height: theme.spacing(6),
     width: theme.spacing(6),
     padding: 0,
+    margin: `0 ${theme.spacing(1)}px`,
+  },
+  settingsIcon: {
+    fill: '#fff',
   },
 }));
 const navItems = [
@@ -37,12 +41,12 @@ export default function SettingsMenu() {
   return (
     <Fragment>
       <IconButton
-        className={classes.avatar}
+        className={classes.settings}
         aria-haspopup="true"
         aria-controls="user-menu"
         aria-label="user menu"
         onClick={handleMenuOpen}>
-        <Settings />
+        <SettingsOutlineIcon size={24} className={classes.settingsIcon} />
       </IconButton>
       <Menu
         id="user-menu"
diff --git a/src/components/Dashboard/index.jsx b/src/components/Dashboard/index.jsx
index 816dbb4..68843d3 100644
--- a/src/components/Dashboard/index.jsx
+++ b/src/components/Dashboard/index.jsx
@@ -23,9 +23,6 @@ const useStyles = makeStyles(theme => ({
     flex: 1,
     justifyContent: 'flex-end',
     alignItems: 'center',
-    '& a button, & a:not(:last-child)': {
-      marginRight: theme.spacing(1),
-    },
   },
   link: {
     textDecoration: 'none',
```

Before:

Screen Shot 2019-05-15 at 3 09 46 PM

After:

Screen Shot 2019-05-15 at 3 02 24 PM

File: src/components/Dashboard/index.jsx [hunk 3]
```diff
@@ -33,6 +34,10 @@ const useStyles = makeStyles(theme => ({
   buttonWithIcon: {
     paddingLeft: theme.spacing(2),
   },
+  sidenav: {
```
Comment (helfi92) [inline src/components/Dashboard/index.jsx]: I think `sidenav` is unused.

File: src/views/History/index.jsx [hunk 4]
```diff
@@ -11,6 +11,9 @@ function History() {
   async function handleFormSubmit(data) {
     const [err, history] = await tryCatch(getHistory(data));
 
+    // shut the linter up for now
```
Comment (helfi92) [inline src/views/History/index.jsx]: ? 😅

File: src/views/RequiredSignoffs/ViewSignoff/index.jsx [hunk 5]
```diff
@@ -43,12 +43,15 @@ export default function ViewSignoff() {
...
 
+  console.log(products);
+  console.log(setProducts);
   useEffect(() => {
     console.log('prior to getting products');
 
     (async () => {
       const [error, result] = await tryCatch(getProducts());
 
+      console.log(error);
```
Comment (helfi92) [inline src/views/RequiredSignoffs/ViewSignoff/index.jsx]: The next PR will display an error component when an error is there. We can remove this line.

File: src/components/Dashboard/menuItems.js [hunk 6]
```diff
@@ -7,14 +7,6 @@ export default [
...
     path: '/releases',
   },
-  {
-    value: 'Permissions',
-    path: '/permissions',
-  },
-  {
-    value: 'Signoffs',
-    path: '/signoffs',
-  },
```
Comment (helfi92) [inline src/components/Dashboard/menuItems.js]: Nitpicking - feel free to ignore this comment

I think we can group all of the links here:

```jsx
export default {
  main: [
    {
      value: 'Rules',
      path: '/rules',
    },
    {
      value: 'Releases',
      path: '/releases',
    },
    {
      value: 'History',
      path: '/history',
    },
  ],
  settings: [
    {
      value: 'Users',
      path: '/users',
    },
    {
      value: 'Roles',
      path: '/roles',
    },
    {
      value: 'Required Signoffs',
      path: '/required-signoffs',
    },
  ],
};
```

**Comment by helfi92** [comment]
Fixes https://github.com/mozilla-frontend-infra/balrog-ui/issues/5.

File: src/components/Dashboard/SettingsMenu.jsx (hunk 2 above)
Comment (mozbhearsum) [reply src/components/Dashboard/SettingsMenu.jsx]: Much nicer!

File: src/views/History/index.jsx (hunk 4 above)
Comment (mozbhearsum) [reply src/views/History/index.jsx]: `eslint` was throwing some errors about unused variables for me, I just wanted to quiet it down :)

File: src/components/Dashboard/menuItems.js (hunk 6 above)
Comment (mozbhearsum) [reply src/components/Dashboard/menuItems.js]: Good idea, I'll do it.

File: src/views/History/index.jsx (hunk 4 above)
Comment (helfi92) [reply src/views/History/index.jsx]: lol. I think you can make it quiet via

```jsx
// eslint-disable-next-line no-console
console.log('...');
```

**Pull request merged**